### ISSUE TITLE
Donation popup showing up more than once

### ISSUE THREAD

**Issue by tehoko** [saenzramiro/rambox#414]
Rambox is showing a "WE NEED YOUR HELP!" popup **each time it starts**. The popup covers all of the interface, save for the menu buttons at the top, rendering the actual messaging part of Rambox unusable until the popup is dismissed.

We all know money can be an issue, but I don't think this behavior is acceptable. I would be okay if Rambox had shown it **once** - for example, after a week of use - and saved my choice across startups, so that I don't see the popup **ever again**. But showing it on each startup is way too much.

I'm certainly not alone with this issue. [You can see here](https://github.com/evertheylen/rambox/commit/a431f12cbedd270662a1281d4703369b6adb0035) that another user decided to uncomment the popup, because it's so annoying.

Rambox is open source, so anyone can do what @evertheylen did and compile the code themselves. But is that what you aim for? Would an average non-donating user do that, or would they just seek a less annoying alternative?

I don't even know if the popup will actually stop appearing if I donate. Maybe I need to be logged in to a Rambox account? Maybe it will, but on another PC I will have to go through all this again? It's not explained anywhere in the popup or the rest of the application.

What I *do* know, though, is that I'm not entitled to anything for free, and you're free to leave the popup intact. I'd just like to know if that'll ever change, or if it's time to move on.

> Rambox: v0.4.4

**Comment by evertheylen** [comment]
TBH I don't actually use my own fork (yet). I try to keep my packages organized (using Arch Linux), and wasn't able to quickly write a PKGBUILD for it. The standard package in the AUR uses binaries. But yeah, I agree. I **totally** understand that the maintainer wants to earn some money for his effort, but whenever open source projects start aggressively asking for money they don't usually end well.

**Comment by gnidorah** [comment]
I believe this started with v0.4.4

**Comment by brwolfgang** [comment]
I even donated hoping the constant popup would go away - not something I regret, I use Rambox all day everyday - but after all this time I'm really tired of this popup. I strongly agree with @tehoko here, the popup must be less frequently shown.

**Comment by saenzramiro** [comment]
Ok, so I configured to only displays once.

But what you propose to incentive people to donate?

**Comment by brwolfgang** [comment]
How about a banner on top of each and every service from time to time telling them to? The banner would come and go automatically and would not appear very frequently. The most irritating part about the popup approach is that it's a modal popup, you can't interact with the app until you close it. 
The banner would not be modal and would not take a lot of screen space - also, it would appear for just a few seconds while the window is active to make sure the user sees it -, and if you choose the colors right it wouldn't bother people at all.
Well, just a raw idea, but I believe it's gonna work.

**Comment by MrChick** [comment]
I'd suggest a banner at the bottom of rambox which shows up occasionally. Nothing too big, something like a one line height bar with some text like "rambox is free software, please consider donating to support its development" or so.

**Comment by saenzramiro** [comment]
Ok, I will consider it.

**Comment by 0mega42** [comment]
How about you change the mostly useless loading screen to a big "this is rambox! click here to donate" screen?

**Comment by hombrepac** [comment]
@saenzramiro No banners or popups please. In my case it actually causes the opposite to the desired effect, and I'm quite sure a lot of the early users and open source supporters don't enjoy having those kind of screens or banners thrown at their faces. Most of us actually use popups and ad blockers on our browsers to avoid that kind of behavior. 

I understand the need for donations, but please remember that unobtrusive publicity (or requests for donations) always are better received. 

@0mega42 suggestion is a really nice option. The loading screen + some kind of marquee (go 90's!), maybe rotating every few minutes down under where the "coded with love from Argentina" is located (or to the right of the services tabs,if there is free space), while making the donation button with the heart "glow" on the bottom... or something along those lines. In my humble opinion it should not interrupt the workflow or require an action to remove it.

**Comment by lordbertus** [comment]
Yes, I agree this popup is the only thing that would keep me from recommending Rambox as otherwise it works great for me. I also think some kind of less intrusive or one-time message would be a lot more acceptable. As well as tehoko remarked it would be better if the message doesn't show for people who already donated. I have no problem donating for a useful and open source app, but if it keeps asking for donations that kinda defeats the point.

**Comment by nmat** [comment]
based on @hombrepac suggestion.

Might be a good idea to just have a step by step introduction using the same size of the screen. Like a getting started guide and then at the end have a "Do you like Rambox, please consider donating. You can reach all information under Rambox settings dialog in app."

And have that only run once maybe?

**Comment by saenzramiro** [comment]
Getting started guide it's a good option. The thing is that I use Tooltip.io to display modals and banners dynamically and there is an option to display once to the user. I always select that for donations but seems not working correctly... sorry guys.

I will consider using another option in the future.

**Comment by daniel-illi** [comment]
A friend of mine maintains a rather popular OSX application, he choose a slightly different approach.

The application shows a modal dialog one time after the first startup for every major update. There he lists the feature changes and also shows the donation information.

This seems to give a good balance between "nag for donations" and "benefit for the user". There are no significant complaints from the users and he gets a nice sum of donations after every major release.

Might be worth to try?
